client/disputes/redirect-to-transaction-details.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies.
+ */
+import { useDispute } from 'data/index';
+import Page from 'components/page';
+import { Dispute } from 'wcpay/types/disputes';
+import { Charge } from 'wcpay/types/charges';
+import { getAdminUrl } from 'wcpay/utils';
+
+const RedirectToTransactionDetails = ( {
+	query: { id: disputeId },
+}: {
+	query: { id: string };
+} ): JSX.Element => {
+	const { dispute, isLoading } = useDispute( disputeId );
+	const disputeObject = dispute || ( {} as Dispute );
+	const disputeIsAvailable = ! isLoading && dispute && disputeObject.id;
+	// Why would dispute.charge ever be a string?
+	const chargeObject = disputeObject.charge as Charge;
+
+	let transactionDetailsUrl = '';
+	if ( disputeIsAvailable ) {
+		const paymentIntentId = disputeObject.charge;
+		transactionDetailsUrl = getAdminUrl( {
+			page: 'wc-admin',
+			path: '/payments/transactions/details',
+			id: chargeObject.payment_intent,
+			transaction_id: chargeObject.balance_transaction,
+			type: 'dispute',
+		} );
+		// window.location = transactionDetailsUrl;
```

Comment:
To improve the UX, instead of a full-page reload, I'd recommend we use `@woocommerce/navigation` client-side routing, similar to the Transaction Details screen redirect.

https://github.com/Automattic/woocommerce-payments/blob/ee35b7685964a53db150de342d1f690598f2e736/client/payment-details/charge-details/index.tsx#L49-L59

Reply:
Nice – I can add that in right now to get this closer.

I think there will still be some slowness experienced by the user, as we need to pull the dispute (AJAX). That's ok though I wonder if we can avoid it. 

I guess that will get addressed by other PRs fixing the links elsewhere in admin, so maybe not a concern in practice.

Reply:
Added here: 1409a1050c5273e21aeb13c00627a90948bf228f

The redirect is now working well, just need a little interstitial UI.